modules/tribes/tests/client/components/TribesPage.component.test.js

```diff
@@ -0,0 +1,177 @@
+/* tests to write
+
+when clicking join and not logged in, redirect to tribe page
+when clicking join and logged in, send a request to join to API
+when clicking join and logged in, update the number of tribe members
+when clicking leave, send a request to leave to API
+when clicking leave and logged in, update the number of tribe members
+
+
+*/
+
+import React from 'react';
+import * as rtl from '@testing-library/react';
+import { render, fireEvent, wait } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import '@/config/client/i18n';
+
+import TribesPage from '@/modules/tribes/client/components/TribesPage.component';
+import * as api from '@/modules/tribes/client/api/tribes.api';
+
+jest.mock('@/modules/tribes/client/api/tribes.api');
+
+const dummyTribes = [{
+  _id: 'aaaa',
+  id: 'aaaa',
+  count: 20000,
+  slug: 'hitchhikers',
+  label: 'Hitchhikers',
+}, {
+  _id: 'bbbb',
+  id: 'bbbb',
+  count: 5000,
+  slug: 'snails',
+  label: 'Snails',
+}, {
+  _id: 'cccc',
+  id: 'cccc',
+  count: 1500,
+  slug: 'teddybears',
+  label: 'Teddy Bears',
+}];
```

Comment:
I think hard coded values are fine for now, but I kind of like a faker library that generates some random nuimbers/text, just in case the tests are accidentally hard coding some expectations based on static data.

Reply:
I like that, although I haven't used it before. Do you know a library you would recommend @nicksellen? Maybe the `faker` library we already have in our `devDependencies` could be used for this purpose?

Reply:
I'm trying to do this with the `faker` library, but I suggest we do it in different PR. It's a lot of added logic. WDYT?

Reply:
I on the other hand like when the test data is always the same & consistent on each run but has variety within tests to cover odd cases. 😁  That way test runs don't have random flakiness. Not a strong opinion tho!